cmd/axelard/cmd/vald/evm/rpc/rpcClient.go

```diff
@@ -13,7 +13,11 @@ import (
 
 //go:generate moq -out ./mock/rpcClient.go -pkg mock . Client MoonbeamClient
 
-const codeMethodNotFound = -32601
+const (
+	codeMethodNotFound = -32601
+	codeServerError    = -32000
+	ganacheChainID     = int64(1337)
```

Comment:
Yeah, I'm not liking this approach of looking for specific error codes since there are inconsistencies in implementation. Since @sammy1991106 didn't want to rely on checking if the chain name in config was `moonbeam` for generality, maybe instead of checking for error codes, we flip the check. If the call to `chain_getFinalizedHead` succeeds, then we treat the client as moonbeam. If there's any error returned, then we treat the RPC as a general evm client. Since we're already sanity checking if the RPC connection can be made via the `Dial` call before, I think this approach should be more stable rather than relying on the error code type.
Another solution would be to combine these two suggestions: Check for success. If this call fails, but the chain name is moonbeam, then return an error. This way we have a generic solution, but also prevent someone setting a bad RPC for moonbeam.

Reply:
Also, agree with @ggutoski here. A ganache's hardcoded chain ID should not exist in our code. As a matter of fact, this is only the default chain ID that ganache uses and one can configure to change it.